server/event/dispatch/operation.go

```diff
@@ -187,7 +187,7 @@ func (o *Operation) evaluateStringExpression(statement string, errorInfo string)
 	return v, nil
 }
```

Comment:
I guess I should drop the `Selector` matching logic in `dispatch`?

https://github.com/argoproj/argo-workflows/blob/21363e524868d6b0e0a9730de81b08db724d79dd/server/event/dispatch/operation.go#L75-L82

Unless we want to keep both validations.